connect/api/src/main/java/org/apache/kafka/connect/rest/ConnectRestExtensionContext.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.rest;
+
+import org.apache.kafka.connect.health.ConnectClusterState;
+
+import javax.ws.rs.core.Configurable;
+
+/**
+ * The interface provides the ability for {@link ConnectRestExtension} implementations to access the JAX-RS
+ * {@link javax.ws.rs.core.Configurable} and cluster state {@link ConnectClusterState}. The implementation for the interface is provided
+ * by the Connect framework.
+ */
+public interface ConnectRestExtensionContext {
+
+    /**
+     * Provides an implementation of {@link javax.ws.rs.core.Configurable} that be used to register JAX-RS resources.
+     *
+     * @return @return the JAX-RS {@link javax.ws.rs.core.Configurable}; never {@code null}
+     */
+    Configurable<? extends Configurable> configurable();
+
+    /**
+     * Provides the cluster state and health information about the connectors and tasks.
+     *
+     * @return the cluster state information; never {@code null}
+     */
+    ConnectClusterState clusterState();
```

Comment:
added some docs on ConnectClusterState. Let me know

Reply:
With the changes to externalize secrets in [KIP-297](https://cwiki.apache.org/confluence/display/KAFKA/KIP-297%3A+Externalizing+Secrets+for+Connect+Configurations), the existing ConnectClusterState is going to output the transformed configurations, which may include secrets. Do we want to do that? Or, would we prefer to get the pre-transformed configurations?

Reply:
IIUC, KIP-297 is for connector configurations and we are only dealing with those here atm.

Reply:
Sorry, I now realize that this `ConnectClusterState` is the interface in the API, whereas KIP-297 is changing the implementation in the runtime. The API interface doesn't expose the configuration, so that's a good thing.

Reply:
Sorry, I now realize that this ConnectClusterState is the interface in the API, whereas KIP-297 is changing the implementation in the runtime. The API interface doesn't expose the configuration, so that's a good thing.

Reply:
I actually think the question is still relevant -- `ConnectClusterState` used to be purely immutable and now we'll be exposing an interface that changes based on when you call it. I think it doesn't matter much here, but it is mainly relevant because the docs on `ConnectClusterState` aren't really accurate anymore since the contents can change over time. But it's also a weird mix of mutability -- the set of connectors & tasks won't change in the `ConnectClusterState` object, you would need to re-callthis method to get updated cluster state. However, the actual values returned for a connector/task config *could* change due to KIP-297 replacements.